cache/client.go

```diff
@@ -0,0 +1,122 @@
+package cache
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/types"
+	"github.com/NebulousLabs/merkletree"
+)
+
+func verifyBlockHeader(header types.BlockHeader, minTimestamp types.Timestamp) error {
+	// Check that the timestamp is not too far in the past to be acceptable.
+	if minTimestamp > header.Timestamp {
+		return fmt.Errorf("Block header validation failed: EarlyTimestamp")
+	}
+
+	// Check if the block is in the extreme future. We make a distinction between
+	// future and extreme future because there is an assumption that by the time
+	// the extreme future arrives, this block will no longer be a part of the
+	// longest fork because it will have been ignored by all of the miners.
+	if header.Timestamp > types.CurrentTimestamp()+types.ExtremeFutureThreshold {
+		return fmt.Errorf("Block header validation failed: ExtremeFutureTimestamp")
+	}
+
+	// Check if the block is in the near future, but too far to be acceptable.
+	// This is the last check because it's an expensive check, and not worth
+	// performing if the earlier checks failed.
+	if header.Timestamp > types.CurrentTimestamp()+types.FutureThreshold {
+		return fmt.Errorf("Block header validation failed: FutureTimestamp")
+	}
+	return nil
+}
+
+// minimumValidChildTimestamp returns the earliest timestamp that a child node
+// can have while still being valid.
+func minimumValidChildTimestamp(headers []types.BlockHeader, headerIndex int) (minTimestamp types.Timestamp, err error) {
+	// Get the previous MedianTimestampWindow timestamps.
+	windowTimes := make(types.TimestampSlice, types.MedianTimestampWindow)
+	windowTimes[0] = headers[headerIndex].Timestamp
+	parent := headers[headerIndex].ParentID
+	for i := 1; i < int(types.MedianTimestampWindow); i++ {
+		// If the genesis block is 'parent', use the genesis block timestamp
+		// for all remaining times.
+		if parent == (types.BlockID{}) {
```

Comment:
s/(types.BlockID{})/types.BlockID{}/

Reply:
Won't work out because of `if`, we get syntax error without parentheses.

Reply:
I see. `{` is attracted to `if` more than to types.BlockID